### DISCUSSION TITLE
How do I surround a movement with parenthesis in insert-mode?

### DISCUSSION THREAD

**Discussion by gregorias** [kylechui/nvim-surround#225] [Q&A] [answered]
With an insert-mode state like this: `val = foo*arg`. I would like to get `val = foo(arg)`. It would be nice to type a sequence like `<C-G>s$)` for that effect.

I know this is easily achievable in the normal or visual mode, but I wonder if it's possible in insert-mode as well. This kind of situation arises quite often in my workflow.

**Comment by kylechui** [comment]
There currently isn't a mechanism for doing this, although you could try(?) to create your own custom mapping by using the provided `<Plug>` mappings. I haven't spent any time thinking about this, so I'm also not 100% sure on what that would look like or if it's even feasible. However, I will say that for the use case that you present, the `f` surround might help if you started out with `arg` and wanted to surround it with `foo(` and `)`.

**Reply by gregorias** [reply, marked as answer]
Thank you for the idea. I remapped `<C-y>` to run nvim-surround in normal mode once:

```lua
require("which-key").register({
	["<C-y>"] = { "<C-o>ys", "(You) surround with" },
}, { mode = "i", silent = true, noremap = false })
```
